fitbenchmarking/core/tests/test_results_fallback.py

```diff
@@ -0,0 +1,660 @@
+        """
+        actual_column_tags = _find_non_full_columns(columns,
+                                                    expected_count,
+                                                    columns_with_errors)
+        self.assertEqual(actual_column_tags, expected_column_tags)
+
+    def test_find_non_full_columns(self):
+        """
+        Test the _find_non_full_columns function
+        """
+        columns = {'s1:m10:j0:': 2,
+                   's1:m11:j0:': 2,
+                   's0:m01:j0:': 2,
+                   's1:m10:j1:': 2,
+                   's1:m11:j1:': 2,
+                   's0:m01:j1:': 2,
+                   's0:m00:j0:': 1,
+                   's0:m00:j1:': 1}
+        expected_count = 2
+        columns_with_errors = {'s0:m00:[^:]*:[^:]*': 1}
+
+        # Test Case 1
+        # Using the results in checkpoint.json
+        # No column with fallback
+        # 1 Error result
+        expected_column_tags = []
+        self.run_and_match_find_non_full_columns(columns,
+                                                 expected_count,
+                                                 columns_with_errors,
+                                                 expected_column_tags)
+
+        # Test Case 2
+        # Using the results in checkpoint.json
+        # 1 column with fallback
+        # 0 Error result
+        columns_with_errors = {}
+        columns['s0:m00:j0:'] = columns['s0:m00:j1:'] = 2
+        expected_column_tags = []
+        self.run_and_match_find_non_full_columns(columns,
+                                                 expected_count,
+                                                 columns_with_errors,
+                                                 expected_column_tags)
+
+        # Test Case 3
+        # Using the results in checkpoint.json
+        # 2 column with fallback
+        # 0 Error result
+        columns['s1:m11:j0:'] = columns['s1:m11:j1:'] = 1
+        expected_column_tags = ['s1:m11:j0:', 's1:m11:j1:']
+        self.run_and_match_find_non_full_columns(columns,
+                                                 expected_count,
+                                                 columns_with_errors,
+                                                 expected_column_tags)
+
+        # Test Case 4
+        # Using the results in checkpoint.json
+        # 2 column with fallback
+        # 0 Error result
+        columns['s1:m11:j0:h0'] = columns['s1:m11:j1:h1'] = 1
+        expected_column_tags = ['s1:m11:j0:',
+                                's1:m11:j1:',
+                                's1:m11:j0:h0',
+                                's1:m11:j1:h1']
+        self.run_and_match_find_non_full_columns(columns,
+                                                 expected_count,
+                                                 columns_with_errors,
+                                                 expected_column_tags)
+
+    def test_handle_fallback_tags(self):
```

Comment:
What is this unit test testing?
Is it the case where `_handle_fallback_tags` is given no fallbacks?

I think this function could do with multiple unit tests with descriptive names (subtests are fine)

Reply:
This is the original test I wrote for `checkpoint.json`. It was a longer test. I was manipulating `checkpoint.json ` for another scenario. But that scenario is now covered by one of the edge cases. 

I think I can rename the TestEdgeCase class as you mentioned in a later comment for a more descriptive name and merge the two classes now.

`checkpoint.json` is a test with no fallback columns but it has a result with error_flag = 4. None of the other edge cases have error_flag = 4. It is useful for testing `_find_non_full_columns()` and `_get_all_result_tags()`. So that is why that test existed but I think I can merge the test into one function that tests all the files.